### PR TITLE
fix: expression only choices / select options

### PR DESCRIPTION
**Pull request by taozhou-glean**
# Description

to fix https://github.com/lingui/js-lingui/issues/1522

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

Fixes # (issue) https://github.com/lingui/js-lingui/issues/1522

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/macro/src/macroJs.ts [hunk 1]
```diff
@@ -352,8 +352,12 @@ export default class MacroJs {
           value = this.tokenizeTemplateLiteral(attrValue)
         } else if (this.types.isCallExpression(attrValue)) {
           value = this.tokenizeNode(attrValue)
+        } else if (this.types.isStringLiteral(attrValue)) {
+          value = attrValue.value
+        } else if (this.types.isExpression(attrValue)) {
```
Comment (taozhou-glean) [inline packages/macro/src/macroJs.ts:357]: StringLiteral is also considered as Expression, so I moved StringLiteral check above. From ts,  after all the checks, the types fall through are limited to `babelTypes.RestElement | babelTypes.AssignmentPattern | babelTypes.ArrayPattern | babelTypes.ObjectPattern`,  not sure exactly how we want to handle them, so I kept the previous fallback case untouched.

Comment (taozhou-glean) [reply packages/macro/src/macroJs.ts:357]: the `unknown` conversion is just to make ts happy, please let me know if you have other suggestions to handle the remaining cases.

File: packages/macro/src/macroJs.test.ts [hunk 2]
```diff
@@ -262,6 +262,47 @@ describe("js macro", () => {
...
+            name: "drink",
+            value: expect.objectContaining({
+              name: "drink",
+              type: "Identifier",
+            }),
+          },
+        },
+      })
+    })
+
```
Comment (thekip) [inline packages/macro/src/macroJs.test.ts:305]: You can safely drop unit tests from macroJs.test.ts file. They don't have a lot of sense, it's perfectly covered from fixtures with js code and this just create more work for maintenance.

**Review by thekip — CHANGES_REQUESTED** [review]
I think the same is true for JSX <Plural> and <Select> components. Could you also check add that? 

And it also should be checked for SWC Plugin. I think it should work already but at least the same test cases should be added to have test suite aligned.

**Comment by taozhou-glean** [comment]
swc-plugin change in https://github.com/lingui/swc-plugin/pull/40 

is there a guide how to run tests etc ? I have zero experience with rust before... I can read more about it but would be great if there is a command that I can just copy and paste to run the tests ;)

**Review by thekip — APPROVED** [review]

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**